build.go

```diff
@@ -200,35 +191,49 @@ func (p spkParams) goEnv() []string {
 // The outer tar file is uncompressed and contains the minimal
 // metadata. The main contents are in the outer tar's "package.tgz"
 // entry, which is gzip or xz compressed.
-func genSPK(w io.Writer, param spkParams) error {
-	privFile := fmt.Sprintf("privilege-dsm%d", param.dsm)
-	if param.forPackageCenter {
-		privFile += ".priv"
-	}
+func genSPK(param spkParams) error {
+	for _, dsm := range param.dsms {
```

Comment:
I don't like this loop. It doesn't match the function name or the godoc for the func which are all about generating a single SPK.

I'd rather see a bunch of small functions that are individually readable (and testable in theory) rather than long funcs with loops of loops and logic sprinkled around (you also have a loop over synoArch below!)

Reply:
how's this?